company.rb

```diff
@@ -0,0 +1,65 @@
+$LOAD_PATH << '.'
+require 'quality-assurance'
+require 'human-resource'
+require 'employee'
+require 'module'
+
+
+class Company
+include Read_CSV_File
+  attr_accessor :company_id, :company_name, :company_type, :headquarter, :founded
+  @@no_of_employees = 0
+  
+
+  def initialize(company_id, company_name, company_type, headquarter, founded)
+    @company_id = company_id
+    @company_name = company_name
+    @company_type = company_type
+    @headquarter = headquarter
+    @founded = founded
+    hr_object = HR.new(0o01, 'HR', 'recruitment, selection, placement, socialisation')
+    qa_object=QA.new(0o02, 'QA', 'Good')
+    @departments = [hr_object,qa_object]
+  end
+
+  def show_details
+    puts "Here's the information for #{company_name}"
+    puts "Company Name - #{company_name}"
+    puts "Company Type - #{company_type}"
+    puts "Number of employees - #{@@no_of_employees}"
+    puts "Headquarter - #{headquarter}"
+    puts "Founded in - #{company_name}"
+  end
+
+  def create_employee(employee_hash)
+    @dept_id=search_department(employee_hash['department_key'])
+     unless @dept_id.nil?
+      empobj=Employee.new(employee_hash['employee_id'],employee_hash['employee_name'],employee_hash['age'],employee_hash['birth_date'],employee_hash['address'],@dept_id)
```

Comment:
Should be `empObj = Employee.new(employee_hash)`

# dependency inversion. Company class should not know the details of the employee object. 
If you need to add one more detail in the future, you will have to update this company class as well if you pass every employee key.